modules/xray/shared/src/main/scala/natchez/xray/XRayEnvironment.scala

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2019-2020 by Rob Norris and Contributors
+// This software is licensed under the MIT License (MIT).
+// For more information see LICENSE or https://opensource.org/licenses/MIT
+
+package natchez.xray
+
+import cats.data._
+import cats.effect._
+import com.comcast.ip4s.{IpAddress, SocketAddress}
+import natchez.Kernel
+
+trait XRayEnvironment[F[_]] {
+  def daemonAddress: F[Option[SocketAddress[IpAddress]]]
+  def traceId: F[Option[String]]
+  def kernelFromEnvironment: F[Kernel]
+}
+
+object XRayEnvironment extends XRayEnvironmentCompanionPlatform {
+  def apply[F[_]: XRayEnvironment]: XRayEnvironment[F] = implicitly
+
+  implicit def instance[F[_]: Sync]: XRayEnvironment[F] =
+    new XRayEnvironment[F] {
+      override def daemonAddress: F[Option[SocketAddress[IpAddress]]] =
+        OptionT(Sync[F].delay(env.get("AWS_XRAY_DAEMON_ADDRESS")))
+          .subflatMap(SocketAddress.fromStringIp)
+          .value
+
+      override def traceId: F[Option[String]] =
+        Sync[F].delay(env.get("_X_AMZN_TRACE_ID"))
+
+      override def kernelFromEnvironment: F[Kernel] =
+        OptionT(traceId)
+          .map(XRaySpan.XRayHeader(_, None, sampled = true).toKernel)
```

Comment:
I think I got this wrong in my pull request. It looks like `traceId` returns something like `Root=1-619446d4-2de198a61485ba3d50bf889d;Parent=7c143d692eca32b3;Sampled=1` if `_X_AMZN_TRACE_ID` is present in the environment, meaning the value should be parsed instead of just dropped into a new `XRayHeader` directly.

Instead, I think this works:

```scala
OptionT(traceId)
  .subflatMap(XRaySpan.parseHeader)
  .map(_.toKernel)
  .getOrElse(Kernel(Map.empty))
```